__tests__/app.test.js

```diff
@@ -166,56 +154,96 @@ describe('/api/articles/:article_id/comments', () => {
+      .get('/api/articles/999/comments')
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Comments does not exist');
+      });
+  });
+  test('The \'/api\' endpoint to include a description of this new \'/api/articles/:article_id/comments\' endpoint.', () => {
+    return request(app)
+      .get('/api')
+      .set('Accept', 'application/json')
+      .expect('Content-Type', /json/)
+      .expect(200)
+      .then(({ body }) => {
+        expect(
+          body.endpoints['GET /api/articles/:article_id/comments'].exampleResponse)
+          .toBeInstanceOf(Object);
+        expect(
+          body.endpoints['GET /api/articles/:article_id/comments'].exampleResponse)
+          .toEqual(
+            endpoints['GET /api/articles/:article_id/comments'].exampleResponse);
+      });
+  });
+});
+describe('POST /api/articles/:article_id/comments', () => {
+  test('POST:200 should insert new comment to DB and return it', () => {
+    return request(app)
+      .post('/api/articles/1/comments').send({ username: 'butter_bridge', body: 'a cat' })
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.comment).toBeInstanceOf(Object);
+        expect(typeof body.comment.comment_id).toBe('number');
+        expect(typeof body.comment.body).toBe('string');
+        expect(typeof body.comment.article_id).toBe('number');
+        expect(typeof body.comment.author).toBe('string');
+        expect(typeof body.comment.votes).toBe('number');
+        expect(typeof body.comment.created_at).toBe('string');
+      });
+  });
+  test('POST:400 sends an appropriate status and error message when given an invalid article_id', () => {
+    return request(app)
+      .post('/api/articles/not-valid-id/comments').send({ username: 'butter_bridge', body: 'a cat' })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Bad Request');
+      });
+  });
+  test('POST:500 sends an appropriate status and error message when given an invalid username', () => {
+    return request(app)
+      .post('/api/articles/1/comments').send({ username: 'this_user_name_doesn\'t_exist', body: 'I\'m not real user' })
+      .expect(500)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Internal Server Error');
+      });
+  });
+  test('The \'/api\' endpoint to include a description of this new POST \'/api/articles/:article_id/comments\' endpoint.', () => {
+    return request(app)
+      .get('/api')
+      .set('Accept', 'application/json')
+      .expect('Content-Type', /json/)
+      .expect(200)
+      .then(({ body }) => {
+        expect(
+          body.endpoints['POST /api/articles/:article_id/comments'].exampleResponse)
+          .toBeInstanceOf(Object);
+        expect(
+          body.endpoints['POST /api/articles/:article_id/comments'].exampleResponse)
+          .toEqual(
+            endpoints['POST /api/articles/:article_id/comments'].exampleResponse);
+      });
+  });
 });
```

Comment:
There's a couple of tests missing here for the POST /api/articles/:article_id/comments:
* Status 404 - trying to add a new post to an article_id which does not exist (i.e. /api/articles/99999/comments)
* Status 201 - what happens if the client tries to add properties to the `.send()` which the server doesn't use? it should ignore them, for example if the client sends `{ username: 'butter_bridge', body: 'a cat', anotherKey: "ignore-me" }` should be successful
* Status 400 - what happens if the client posts a new comment but is missing vital information which the request needs, for example they don't send the `username` on the object?